airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -154,6 +177,7 @@ def current_state(self, customer_id, default=None):
 
     def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[MutableMapping[str, any]]]:
         for customer in self.customers:
+            logger = cyclic_sieve(self.logger, 10)
```

Comment:
Can we avoid initializing a new `cyclic_sieve` since it should have already been defined on the stream as `self.incremental_sieve_logger`?

Reply:
Technically we can, but this was done on purpose. The idea was to filter out not just 9 out of 10 messages, but 9 out of 10 message cycles to have more consistent logs so an engineer shouldn't have to stick together messages from different routine calls / cycles / exceptions / if statements and so on when debugging. Since `stream_slices()` is called once per stream before reading records, and `read_records()` is called once per stream slice, these are two different log cycles.
If you still insist, I can reuse `self.incremental_sieve_logger` here, technically it is not a problem

Reply:
Ah got it, that makes sense and I agree that for a better experience we wouldn't want to share the same log cycles. Can you add a comment for that here just mentioning that's why a logger sieve is initialized here.